internal/local/app_test.go

```diff
@@ -698,6 +698,32 @@ exports = function({ query }) {
 				},
 			},
 		},
+		{
+			Config: map[string]interface{}{
+				"type":    "mongodb",
+				"name":    "mdbSvc",
+				"config":  map[string]interface{}{},
+				"version": float64(1),
+			},
+			DefaultRule: map[string]interface{}{
+				"roles": []interface{}{
+					map[string]interface{}{
+						"name": "owner",
+						"apply_when": map[string]interface{}{
+							"userId": "%%user.id",
+						},
+						"read": true,
+					},
+				},
+			},
+			Rules: []map[string]interface{}{
+				{
+					"database":   "foo",
+					"collection": "bar",
+					"name":       "foo.bar",
```

Comment:
[q] do we need to specify the `"name"` field here. Technically rule names should only be internal - externally we specify the `database` and `collection` which will eventually get concatenated into a `name`

Reply:
That's what I thought and it's what we do for data sources in v2 of the app structure with v2 rules, but for legacy services with rules and mongo services in the v1 structure it does appear that we use the `"name"` field here: 
https://github.com/10gen/realm-cli/blob/3795c0740d25d61bed4de823807e7351fcd06bbf/internal/local/structure.go#L382

this seems to be intentional but I'll defer to someone with historical context (perhaps @makesitgo) on whether this is how things should be handled for v2 rules in the v1 app structure. if not, it would be a fairly simple change to construct the name for a v2 rule using the database and collection names when writing the rule file.

Reply:
we actually don't seem to ever export `"name"` as a field in any exported rule file for any config version.  see:

* `20200603` and below: https://github.com/10gen/baas/blob/master/importexport/mongodbrules/v2/rule.go#L232-L264
* `20210101` and above: https://github.com/10gen/baas/blob/master/importexport/mongodbrules/v2/rule.go#L312-L328

I think what you're seeing in `local/structure` for the `writeServices` codepath is actually a bug, and we should instead being computing "name" via "database"+"collection" fields.

fortunately (and unfortunately), that codepath is actually never hit in the CLI... I don't love that we needed to go down the route of teaching the CLI how to write _any_ app config files b/c for 98% of the scenarios, the "writing" occurs just by reading the zip response returned from the API.  for context, we went this route so that `app init`/`app create` could link clusters/data sources

Reply:
Thanks for the context! I'll make the change 👍🏽